lib/push.js

```diff
@@ -1,42 +1,76 @@
 var assert = require('assert'),
   futils = require('./fhutils'),
-  aeroGearLoaded = false;
+  xtend = require('xtend'),
+  AeroGear = require('unifiedpush-node-sender');
 
 module.exports = function (cfg) {
   assert.ok(cfg, 'cfg is undefined');
+
   var fhutils = new futils(cfg);
+  var millicoreProps = fhutils.getMillicoreProps();
+  var defaultPushSettings = getPushSettings(millicoreProps);
 
-  var props = fhutils.getMillicoreProps();
-  var headers = {
-    'X-Project-Id': props.widget,
-    'X-App-Id': props.instance
-  };
-  fhutils.addAppApiKeyHeader(headers);
-  var settings = {
-    url: 'https://' + props.millicore + ':' + props.port + '/box/api/unifiedpush/mbaas/',
-    applicationId: "fake", // we have to use fake ID, it will be added by supercore
-    masterSecret: "fake", // we have to use fake secret, it will be added by supercore
-    headers: headers
-  };
+  /**
+   * Generates settings to the used when creating an AeroGear.sender.
+   * @param  {Object} opts [description]
+   * @return {Object}
+   */
+  function getPushSettings (opts) {
+    assert.ok(opts, 'opts is undefined');
 
+    var headers = {
+      'X-Project-Id': opts.widget,
+      'X-App-Id': opts.instance
+    };
+
+    fhutils.addAppApiKeyHeader(headers, opts.appapikey);
+
+    return {
+      url: 'https://' + opts.millicore + ':' + opts.port + '/box/api/unifiedpush/mbaas/',
+      applicationId: "fake", // we have to use fake ID, it will be added by supercore
+      masterSecret: "fake", // we have to use fake secret, it will be added by supercore
+      headers: headers
+    };
+  }
+
+
+  /**
+   * Creates a push client (aka the $fh.push API)
+   * @param  {Object}   settings
+   * @return {Function}
+   */
+  function getPushClient (settings) {
+    var sender = AeroGear.Sender(settings);
```

Comment:
can we keep the existing way of lazy-loading the AeroGear push module. I know it may not be the best practice, but it will reduce the memory usage of a cloud app if push is not used at all.